app/scripts/modules/kubernetes/src/rawResource/component/K8sResources.tsx

```diff
@@ -0,0 +1,111 @@
+import { Application, ApplicationDataSource } from '@spinnaker/core';
+import React from 'react';
+import { FiltersPubSub } from '../controller/FiltersPubSub';
+import { KUBERNETS_RAW_RESOURCE_DATA_SOURCE_KEY } from '../rawResource.dataSource';
+import { RawResource } from './group/RawResource';
+import { RawResourceGroups } from './group/RawResourceGroups';
+import { IK8sResourcesFiltersState } from './K8sResourcesFilters';
+
+export interface IK8sResourcesProps {
+  app: Application;
+}
+
+interface IK8sResourcesState {
+  groupBy: string;
+  filters: IK8sResourcesFiltersState;
+  rawResources: IApiKubernetesResource[];
+}
+
+export class K8sResources extends React.Component<IK8sResourcesProps, IK8sResourcesState> {
```

Comment:
Not a blocker by any means, but we are generally trying to use functional React components instead of class components going forward.